diff.go

```diff
@@ -329,15 +333,17 @@ func diffComponentVolumes(oldDef, newDef ComponentDefinition, componentName Comp
 	return diffInfos
 }
 
-func diffComponentArgs(oldDef, newDef ComponentDefinition, componentName ComponentName) []DiffInfo {
-	diffInfos := []DiffInfo{}
+func diffComponentArgs(oldDef, newDef ComponentDefinition, componentName ComponentName) DiffInfos {
+	diffInfos := DiffInfos{}
 
+	// TODO this needs to be more fine grained
```

Comment:
I mean that we need more detailed diff types for various actions that can take place.